steps/evaluation.py

```diff
@@ -35,13 +35,14 @@
     save_interaction_rdm,
     save_parameter_grid_evaluation,
     evaluate_qubit_operator_list as _evaluate_qubit_operator_list,
+    convert_dict_to_qubitop,
 )
 
 
 def get_expectation_values_for_qubit_operator(
     backend_specs: Union[Dict, str],
-    circuit: Union[str, Circuit],
-    qubit_operator: Union[str, SymbolicOperator],
+    circuit: Union[str, Circuit, Dict],
```

Comment:
I think that the whole serialization/deserialization logic should be inside steps and not here.
The only type that we should allow for should be `Circuit`.
Though in reality this will require a bigger refactor anyway so I think it's fine for now.

Reply:
Damn... It is a step already 🤔 
I think I got confused somewhere and this comment is not valid.